src/lib.rs

```diff
@@ -75,7 +75,7 @@ pub trait Field:
     const ONE: Self;
 
     /// Returns an element chosen uniformly at random using a user-provided RNG.
-    fn random(rng: impl RngCore) -> Self;
+    fn random<R: RngCore + ?Sized>(rng: &mut R) -> Self;
```

Comment:
@baloo what's the motivation for this change? Has the recommended way of generalising RNGs changed in `rand_core 0.9`?

Reply:
Also, this change to the trait should be documented in the changelog.

Reply:
Essentially yes.
For the upstream change: https://github.com/rust-random/rand/pull/1593

We've deployed that downstream in RustCrypto:
 - https://github.com/RustCrypto/crypto-bigint/pull/760
 - https://github.com/RustCrypto/traits/pull/1765

This is to be able to pass a `dyn RngCore` object.

Reply:
Incidentally, this reverts a change we deployed in 0.8.0: #34

At the time the motivation was to align the `ff` and `group` APIs to take the same approach; `rand_core` supported either at the time. If `rand_core` only supports this approach now, then I guess that's fine.